### PR TITLE
[DB] Add gridiron football schema

### PR DESCRIPTION
**Pull request by babrar**
- Gridiron schema from https://app.diagrams.net/#G14WGyYeweR_MhtuVwChezzE5OxJPBMTLZ has been added
- Fixed invalid foreign key constraint and incorrect index creation bug in basketball schema
- Updated drop table script to **CASCADE** on table deletion

NOTE: I chose not to consolidate the **gridironrushes** and **gridironkicks** entities into one entity even though they look very similar. Merging them would have resulted in writing slow queries where we distinguish between rushes and kicks based on the possible values in the `result` attribute. Additionally, rushes and kicks represent sufficiently different entities even though their schemas look similar.

### CODE REVIEW

File: db_scripts/create_tables.sql [hunk 1]
```diff
@@ -227,7 +227,119 @@ CREATE TABLE basketballgameends (
...
+CREATE INDEX IF NOT EXISTS gridiron_events_playtype ON gridirongameevents(playType text_ops);
+CREATE INDEX IF NOT EXISTS gridiron_events_period ON gridirongameevents(period int2_ops);
+
+------------------------------------------------------------------
+
+CREATE TABLE gridironrushes (
+    id uuid PRIMARY KEY DEFAULT uuid_generate_v4(),
+    eventId uuid NOT NULL,
+    player CHARACTER VARYING(255) NOT NULL,
+    yard SMALLINT NOT NULL CHECK (yard > 0),
```
Comment (KnotScientific) [inline db_scripts/create_tables.sql:284]: yards can be zero and negative at times

File: db_scripts/create_tables.sql [hunk 2]
```diff
@@ -227,7 +227,119 @@ CREATE TABLE basketballgameends (
...
+
+CREATE INDEX IF NOT EXISTS gridiron_rushes_result ON gridironrushes(result text_ops);
+
+------------------------------------------------------------------
+
+CREATE TABLE gridironkicks (
+    id uuid PRIMARY KEY DEFAULT uuid_generate_v4(),
+    eventId uuid NOT NULL,
+    player CHARACTER VARYING(255) NOT NULL,
+    yard SMALLINT NOT NULL CHECK (yard > 0),
```
Comment (KnotScientific) [inline db_scripts/create_tables.sql:302]: Same thing

Comment (KnotScientific) [inline db_scripts/create_tables.sql:302]: This can stay positive

File: db_scripts/create_tables.sql [hunk 3]
```diff
@@ -227,7 +227,119 @@ CREATE TABLE basketballgameends (
...
+CREATE INDEX IF NOT EXISTS gridiron_kicks_result ON gridironkicks(result text_ops);
+
+------------------------------------------------------------------
+
+CREATE TABLE gridironthrows (
+    id uuid PRIMARY KEY DEFAULT uuid_generate_v4(),
+    eventId uuid NOT NULL,
+    playerThrowing CHARACTER VARYING(255) NOT NULL,
+    playerReceiving CHARACTER VARYING(255) NOT NULL,
+    yard SMALLINT NOT NULL CHECK (yard > 0),
```
Comment (KnotScientific) [inline db_scripts/create_tables.sql:321]: Can be negative and zero

File: db_scripts/create_tables.sql (hunk 1 above)
Comment (babrar) [reply db_scripts/create_tables.sql:284]: 👀 thanks. didn't know

File: db_scripts/create_tables.sql (hunk 2 above)
Comment (babrar) [reply db_scripts/create_tables.sql:302]: ack

File: db_scripts/create_tables.sql (hunk 3 above)
Comment (babrar) [reply db_scripts/create_tables.sql:321]: ack

File: db_scripts/create_tables.sql [hunk 4]
```diff
@@ -318,7 +318,7 @@ CREATE TABLE gridironthrows (
     eventId uuid NOT NULL,
     playerThrowing CHARACTER VARYING(255) NOT NULL,
     playerReceiving CHARACTER VARYING(255) NOT NULL,
-    yard SMALLINT NOT NULL CHECK (yard > 0),
+    yard SMALLINT NOT NULL CHECK,
```
Comment (KnotScientific) [inline db_scripts/create_tables.sql:321]: do we keep the `CHECK`?

Comment (babrar) [reply db_scripts/create_tables.sql:321]: no that's clearly a mistake from sloppy editing.

Comment (babrar) [reply db_scripts/create_tables.sql]: fixed.

**Review by KnotScientific — APPROVED** [review]

**Pull request merged**